packages/cardano-services-client/src/WebSocket.ts

```diff
@@ -1,41 +1,81 @@
+// cSpell:ignore cardano sonarjs utxos
+/* eslint-disable max-depth */
 /* eslint-disable unicorn/prefer-add-event-listener */
 
 import {
   Cardano,
+  ChainHistoryProvider,
   EpochInfo,
   EraSummary,
   HealthCheckResponse,
   NetworkInfoProvider,
+  Paginated,
   Provider,
   ProviderError,
   ProviderFailure,
+  Serialization,
   StakeSummary,
   SupplySummary,
+  TransactionsByAddressesArgs,
+  UtxoByAddressesArgs,
+  UtxoProvider,
   createSlotEpochInfoCalc
 } from '@cardano-sdk/core';
+import { HexBlob, fromSerializableObject } from '@cardano-sdk/util';
 import { Logger } from 'ts-log';
 import { Observable, ReplaySubject, Subject, filter, firstValueFrom, merge } from 'rxjs';
-import { fromSerializableObject } from '@cardano-sdk/util';
 import WebSocket from 'isomorphic-ws';
 
 const NOT_CONNECTED_ID = 'not-connected';
 
 export type AsyncReturnType<F extends () => unknown> = F extends () => Promise<infer R> ? R : never;
 
+export type MetadataDbModel = [string, HexBlob];
+
 export type NetworkInfoMethods = Exclude<keyof NetworkInfoProvider, 'healthCheck'>;
 export type NetworkInfoResponses = { [m in NetworkInfoMethods]: AsyncReturnType<NetworkInfoProvider[m]> };
 
 export interface WSMessage {
+  /** The addresses the client subscribes. */
+  txsByAddresses?: { addresses: Cardano.PaymentAddress[]; lower: Cardano.BlockNo };
+
   /** The client id assigned by the server. */
   clientId?: string;
 
+  /** The error on server while performing a request. */
+  error?: Error;
+
   /** Latest value(s) for the `NetworkInfoProvider` methods.*/
   networkInfo?: Partial<NetworkInfoResponses>;
+
+  /** The request id from client to server. */
+  requestId?: number;
+
+  /** This message is the response to the message with this id. */
+  responseTo?: number;
+
+  /** The server is still syncing. */
+  syncing?: boolean;
+
+  /** The transactions. */
+  transactions?: Cardano.HydratedTx[];
+
+  /** The partial transactions for UTXOs. */
```

Comment:
Do we even need this as a separate API now, given we're syncing the whole transaction history anyway and can derive the UTxO set? If it's easier to leave in place, it would be good to explain why this response is in the shape of a transaction.

Reply:
Actually we are not loading the whole transaction history: to honor the `blockRange.lowerBound` of `ChainHistoryProvider.transactionsByAddresses` is mandatory to avoid re-loading already loaded transactions when a wallet is re-opened; otherwise opening a wallet would be always a _wallet restore_ procedure.

1b6adaefb99 should solve this adding the explanation.